### PR TITLE
Adapt the mod to the new mod layout

### PR DESCRIPTION
**Pull request by abcdefg30**
This reintroduces the OpenRA/OpenRA submodule and comes with the new scripts from
https://github.com/OpenRA/OpenRAModTemplate. I still need to update some of the scripts (line endings, cake, travis), but the general structuring is finished.

### CODE REVIEW

**Comment by Mailaender** [comment]
I assume you need to update https://github.com/OpenRA/ra2/blob/master/.travis.yml#L24 as well with the new paths. Otherwise looks sane on a first look.

File: OpenRA.Mods.RA2/OpenRA.Mods.RA2.csproj [hunk 1]
```diff
@@ -43,16 +43,16 @@
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="OpenRA.Game">
-      <HintPath>dependencies\OpenRA.Game.exe</HintPath>
+      <HintPath>..\engine\OpenRA.Game.exe</HintPath>
```
Comment (pchote) [inline OpenRA.Mods.RA2/OpenRA.Mods.RA2.csproj]: If you change these to project references then you'll be able to use the "Jump to Definition" IDE feature.

File: build.cake [hunk 2]
```diff
@@ -1,119 +1,32 @@
...
-//
-// Example:
-// cake -auto=true -openra-root=~/projects/openra
 
 var target = Argument("target", "default").ToLowerInvariant();
 var configuration = Argument("configuration", "Debug");
 
-var rootDir = Directory(".");
-var depsDir = Directory("./OpenRA.Mods.RA2/dependencies");
+Task("default").Does(() => {
```
Comment (pchote) [inline build.cake]: In the longer term we should implement this functionality in the mod template (using the standard makefile / batch / powershell scripts) so that all mods can benefit.

File: launch-game.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+TEMPLATE_LAUNCHER=$(python -c "import os; print(os.path.realpath('$0'))")
+TEMPLATE_ROOT=$(dirname "${TEMPLATE_LAUNCHER}")
+source ${TEMPLATE_ROOT}/mod.config
+
+MOD_SEARCH_PATHS="./../.."
```
Comment (pchote) [inline launch-game.sh]: This sets the mod search path to the parent directory of the RA2 clone.  This means that if you have a shared source directory (e.g. `~/src/` with subdirectories for e.g. `ra2` and `d2`) then this will pick up the other mods as well as ra2.  This will probably break things in unexpected ways.

Comment (pchote) [inline launch-game.sh]: As painful as it is to break the git history, the best fix here is to move the mod files into a `mods/ra2` directory as done in the template.

File: mod.config [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# The id of the mod to launch by default.  This must exist as a directory in the mods directory
+MOD_ID="ra2"
+
+# Import the default OpenRA mods for use as external mod references
+# in your mod.yaml. Accepts values "True" or "False".
+# WARNING: This setting is provided to simplify early mod development
+# and must be disabled before installers can be generated
+INCLUDE_DEFAULT_MODS="True"
```
Comment (pchote) [inline mod.config]: You should aim to make whatever fixes you need to reset this to "False" in the shortish term.

**Comment by abcdefg30** [comment]
This waits for https://github.com/OpenRA/OpenRAModTemplate/pull/3 for now.

File: launch-game.sh (hunk 3 above)
Comment (Iamgoofball) [inline launch-game.sh]: I'm fairly certain that GitHub retains history when it detects a file move.

Comment (pchote) [inline launch-game.sh]: It doesn't.  See e.g. https://github.com/OpenRA/OpenRA/commits/bleed/OpenRA.Mods.Cnc/Traits/Chronoshiftable.cs.

Comment (GraionDilach) [inline launch-game.sh]: It's weird - blame definitely retains the history. https://github.com/OpenRA/OpenRA/blame/bleed/OpenRA.Mods.Cnc/Traits/Chronoshiftable.cs

Comment (samrocketman) [inline launch-game.sh]: It depends.  Git doesn't track files, per se, but blobs of text.  The history will be retained if, in a single commit, only a rename happens.  However, if you rename and change code in the same commit then it depends on how much you change.  In general renaming files and folders should be kept as its own commit.

**Comment by abcdefg30** [comment]
Updated. It should work now, but doesn't as it can't find the `cnc.dll` for some reason.

**Comment by pchote** [comment]
Your problem is that you have a stray `$cnc: cnc` in your package list (you don't use it?), and `INCLUDE_DEFAULT_MODS` doesn't appear to be working so the utility can't find it.

**Comment by abcdefg30** [comment]
Finally found the time to update this again.

**Comment by pchote** [comment]
I have a mostly done PR (linux/macOS done, just needs the windows scripts) for https://github.com/OpenRA/OpenRAModTemplate/issues/7.  If you can wait for that it would save you from having to add and then immediately remove a submodule.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRAModTemplate/pull/14 has now been merged.

**Comment by alercah** [comment]
In addition to line endings, the bash scripts need to have +x set.

**Comment by samrocketman** [comment]
@alercah 
> In addition to line endings, the bash scripts need to have +x set.

I looked over all of the bash scripts and none of them explicitly call `set -x` nor pass `-x` via the shabang.  Can you elaborate?  `set +x` is the default for `/bin/sh` and `/bin/bash`.

**Comment by alercah** [comment]
I mean the executable bit, not the bash setting.

**Comment by samrocketman** [comment]
Cool,

    find . -type f -name '*.sh' -exec chmod 755 {} +

**Comment by alercah** [comment]
Sure, that should be in this patch though.

**Comment by abcdefg30** [comment]
Finally fixed & updated.

**Comment by abcdefg30** [comment]
Forgot to set `INCLUDE_DEFAULT_MODS` to `False`, fixed.

**Comment by abcdefg30** [comment]
Fixed travis/linux.
